sdk/tables/azure-data-tables/src/tables_clients.cpp

```diff
@@ -1111,3 +1113,32 @@ std::string TableClient::PrepUpdateEntity(
   returnValue += Serializers::UpdateEntity(entity);
   return returnValue;
 }
+
+std::string TableClient::PrepInsertEntity(
+    std::string const& changesetId,
+    Models::TableEntity entity)
+{
+  std::string payload = Serializers::UpdateEntity(entity);
+  std::string returnValue = "--" + changesetId + "\n";
+  returnValue += "Content-Type: application/http\n";
+  returnValue += "Content-Transfer-Encoding: binary\n\n";
+
+  returnValue += "PATCH " + m_url.GetAbsoluteUrl() + "/" + m_tableName + PartitionKeyFragment
+      + entity.GetPartitionKey().Value + RowKeyFragment + entity.GetRowKey().Value + ClosingFragment
+      + " HTTP/1.1\n";
+  returnValue += "Content-Type: application/json\n";
```

Comment:
I see you have this pattern elsewhere, so consider this a broader question.

A lot of this string construction looks like it is building out a request object, with http method, request line, content-type, etc.

Is it feasible to use our `Azure::Core::Http::Request` type to build out this string?
https://github.com/Azure/azure-sdk-for-cpp/blob/7db4f3ac1c98d624373c138ddd19d8c7ff50e17a/sdk/core/azure-core/inc/azure/core/http/http.hpp#L182

Reply:
it's not quite a request it's a mangled string that looks ugly. i'll look into it though